pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -427,38 +534,179 @@ func (l *ListOptionIndexer) validateColumn(column string) error {
 }
 
 // buildORClause creates an SQLite compatible query that ORs conditions built from passed filters
-func (l *ListOptionIndexer) buildORClauseFromFilters(orFilters OrFilter) (string, []any, error) {
-	var orWhereClause string
+func (l *ListOptionIndexer) buildORClauseFromFilters(orFilters OrFilter, dbName string) (string, []any, error) {
 	var params []any
+	clauses := make([]string, 0, len(orFilters.Filters))
+	var newParams []any
+	var newClause string
+	var err error
 
-	for index, filter := range orFilters.Filters {
-		opString := "LIKE"
-		if filter.Op == NotEq {
-			opString = "NOT LIKE"
+	for _, filter := range orFilters.Filters {
+		if isLabelFilter(&filter) {
+			newClause, newParams, err = l.getLabelFilter(filter, dbName)
+		} else {
+			newClause, newParams, err = l.getFieldFilter(filter)
 		}
-		columnName := toColumnName(filter.Field)
-		if err := l.validateColumn(columnName); err != nil {
+		if err != nil {
 			return "", nil, err
 		}
+		clauses = append(clauses, newClause)
+		params = append(params, newParams...)
+	}
+	return strings.Join(clauses, " OR "), params, nil
+}
 
-		orWhereClause += fmt.Sprintf(`f."%s" %s ? ESCAPE '\'`, columnName, opString)
-		format := strictMatchFmt
+//type Filter struct {
+//	Field   []string
+//	Match   []string
+//	Op      Op: one of
+//	Partial bool
+//}
+
```

Comment:
```suggestion
```

Reply:
Right. I was using that as a guide while writing the code. Not needed now.